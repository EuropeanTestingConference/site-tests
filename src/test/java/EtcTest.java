import com.spun.util.ObjectUtils;
import com.spun.util.SystemUtils;
import com.spun.util.io.FileUtils;
import org.apache.http.HttpResponse;
import org.apache.http.client.HttpClient;
import org.apache.http.client.fluent.Content;
import org.apache.http.client.fluent.Request;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.client.methods.RequestBuilder;
import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.DefaultHttpClient;
import org.apache.http.impl.client.HttpClientBuilder;
import org.apache.http.impl.client.SystemDefaultHttpClient;
import org.approvaltests.Approvals;
import org.junit.Test;

import java.io.IOException;
import java.io.InputStream;
import java.net.URL;


public class EtcTest {

    private static final  boolean USE_LOCAL = true;

    @Test
    public void test2016(){
        verify("2016");
    }

    @Test
    public void testSpeakers(){
    	verify("2016/speakers");
    }

    @Test
    public void test2016Schedule(){
        verify("2016/schedule");
    }

    @Test
    public void test2016Topics(){
        verify("2016/topics");
    }
    @Test
    public void test2017(){
    	verify("2017");
    }

    private void verify(String end) {
        String baseUrl = USE_LOCAL ? "http://localhost:4000/" : "http://europeantestingconference.eu/";
        String html = loadWebPage(baseUrl + end);
        html = normalizeHtml(html);
        verifyPage(html);
    }

    private String normalizeHtml(String html) {
        html = html.replaceAll(" +", " ");
        html = html.replaceAll("(?m)^\t+$", "\n");
        html = html.replaceAll("(?m)^ +$", "\n");
        html = html.replaceAll("\n+", "\n");
        html = customNormalizations(html);
        return html;
    }
	private String customNormalizations(String html) {
//		html = html.replaceAll("/snippets/2016/", "/");
//        html = html.replaceAll("fix-anchor\" id=\"", "fix-anchor\" id=\"/speakers/");
//        html = html.replaceAll("class=\"topic\" id=\"", "class=\"topic\" id=\"/topics/");
//        html = html.replaceAll("a href=\"/speakers#", "a href=\"/speakers#/topics/");
		return html;
	}

    private void verifyPage(String response) {
        Approvals.verifyHtml(response);
    }

    private String loadWebPage(String url)
    {
        try {
            return Request.Get(url).execute().returnContent().asString();
        } catch (IOException e) {
            throw ObjectUtils.throwAsError( e );
        }
    }
}
