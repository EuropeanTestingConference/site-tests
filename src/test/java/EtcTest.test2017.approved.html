<!doctype html>
<html class="no-js" lang="en">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
	<title>European Testing Conference 2017 | European Testing Conference 2016</title>
	<meta name="author" content="European Testing Conference"> 
 	<meta property="og:title" content="European Testing Conference 2017 | European Testing Conference" />
 	<meta property="og:site_name" content="European Testing Conference" />
	<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0">
 <meta name="description" content=""/>
 <link rel="shortcut icon" href="/favicon.ico">
 <link href='http://fonts.googleapis.com/css?family=Source+Sans+Pro:400,600,700' rel='stylesheet' type='text/css'>
 <link rel="stylesheet" href="/styles/main.css">
</head>
<body class="">
<nav id="branding" class="navbar navbar-default">
 <div class="container-fluid">
		<!-- Brand and toggle get grouped for better mobile display -->
 <div class="navbar-header">
 <div class="mini-logo-xs visible-xs-inline push-left">
 <a href="/" id="logo-mini" ><img src="/images/logo/etc-mini_50px.png" alt="European Testing Conference -logo"></a>
 </div>
 <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#etc-navbar-collapse-1" aria-expanded="false">
 <span class="sr-only">Toggle navigation</span>
 <span class="icon-bar"></span>
 <span class="icon-bar"></span>
 <span class="icon-bar"></span>
 <span class="icon-bar"></span>
 </button>
 </div>
 <div class="logo col-md-3 col-md-offset-1 navbar-brand hidden-xs">
 <a href="/" id="logo"><img src="/images/logo/etc-logo_130px.png" alt="European Testing Conference logo"></a>
 </div>
 <div class="collapse navbar-collapse" id="etc-navbar-collapse-1">
 	 <div class="col-md-8">
 		<div class="site-nav">
 <ul class="nav navbar-nav">
 </ul>
	 </div>
		<div class="clearfix"></div>
<!--		<div class="buy-ticket">
			<a class="button" target="_blank" href="https://holvi.com/shop/EuroTestingConf/product/307fb905d2067da1cf9c6a68c2e31e33/">Buy Tickets</a>
		</div> -->
	 </div>
 </div><!-- /.navbar-collapse -->
	<div id="navigation" class="row visible-md-block visible-lg-block hidden fixed">
 	 <div class="inner clearfix">
 	 <div class="mini-logo">
 <a href="/" id="logo-mini" ><img src="/images/logo/etc-mini_50px.png" alt="European Testing Conference -logo"></a>
 	 </div>
 	 <ul class="push-right">
		<!-- <li>
			<a class="button buy-ticket " target="_blank" href="https://holvi.com/shop/EuroTestingConf/product/307fb905d2067da1cf9c6a68c2e31e33/">Buy Tickets</a>
		</li> -->
 </ul>
 	</div>
 </div><!-- /.container-fluid -->
</nav>
<header class="site-header ">
<h1 class="text-center"><a href="/">European Testing Conference 2017</a></h1>
<h2>Helsinki, Finland February 8-10th 2017 at Wanha Satama</h2>
</header>
<main class="site-main container" role="main">
<section id="about" class="col-md-12 main-content text-center ">
 <h2>About</h2>
 <p>Testing is important.
This conference is about getting experts and practitioners together to talk, learn and practice the art of testing. We’re looking into advanced new methods into making our testing more effective, as well as enrich our understanding of fundamental methods to grow a stronger community.</p>
<p>We’re tired of conferences that feel more like sales pitches than practitioners sharing the craft.</p>
<p>Come. Learn. Do. Teach. Test.</p>
<div style="font-size:medium">
Get your ticket now! A super-early Bird Ticket (available until end of August) only 496 euros (incl. VAT) for the two-day conference. It's now the cheapest it will be!
</div>
<div style="font-size:xx-large">
<a href="https://holvi.com/shop/ETC2017/">Buy a ticket here</a>
</div>
<div style="font-size:small">(Speakers will have full 3-day ticket, after passing the Call for Proposals)</div>
</section>
<section id="when-where" class="col-md-12 main-content text-center ">
 <h2>When and Where</h2>
 <div class="tile text-tile col-md-3 col-sm-6 col-xs-12">
 <h3>When?</h3>
 <p>February 8-10th 2017 </p>
</div>
<div class="tile image-tile photo-1 col-md-3 col-sm-6 col-xs-12" style="overflow:hidden;padding:0;">
<a data-flickr-embed="true" href="https://www.flickr.com/photos/hannumakarainen/4617448819/" title="Helsinki by the sea"><img src="https://c4.staticflickr.com/4/3226/4617448819_883a32c6db_b.jpg" width="351" height="230" alt="Helsinki by the sea" /></a><script async="" src="//embedr.flickr.com/assets/client-code.js" charset="utf-8"></script> </div>
<div class="tile tile-2 text-tile col-md-3 col-sm-6 col-xs-12 ">
 <h3>Where?</h3>
 <p>in Wanha Satama, Helsinki, Finland.</p>
</div>
<div class="tile tile-2 image-tile photo-2 col-md-3 col-sm-6 col-xs-12" style="overflow:hidden;padding:0;">
<a data-flickr-embed="true" href="https://www.flickr.com/photos/jukk_a/4218272967/in/photolist-7qKKAg-jnywV-jnywi-BbDRd-FxkhaK-9bWBJB-yCbrA-C5cgov-79UCSX-c1dDt1-GEDWBX-bCBz8A-dym4nL-7w3ikz-bR6r9K-dJwUdz-dtkgjg-dJwYji-bVfGD7-7qPFsG-qLivrv-dfNUYv-bX7m9s-cf2Szy-GgNbEL-c5mpiN-dBh4oE-boY5Xd-dJCpY9-6KFFuS-dJwYt2-dyRAQd-4q2a16-GEPssn-cyNXed-dJwUut-dJCmsj-7B425u-dirfuD-dJCm9j-jG4Rr8-dBmbrC-b9BVPr-6kf3uT-6kjeh5-7wkav5-qHkpBa-brws8x-jJaE3e-DeaZii" title="P1010556"><img src="https://c8.staticflickr.com/5/4063/4218272967_3ff79f8ac1_b.jpg" width="351" height="230" alt="P1010556" /></a><script async="" src="//embedr.flickr.com/assets/client-code.js" charset="utf-8"></script>
</div>
<p>
</p>
</section>
<section id="directions" class="col-md-12 main-content text-center ">
 <h2>Latest Buzz</h2>
 <div class="col-md-6 col-xs-12">
 <hgroup>
 <h2>News<br />
 <span></span></h2>
 </hgroup>
 <article class="news">
 <hgroup>
 <h3><a href="/news/Announcing-Keynote-Speakers">Announcing Keynote Speaker for European Testing Conference 2017</a></h3>
 <p class="writer">10 Aug 2016 / Maaret Pyhäjärvi</p>
 </hgroup>
 <p>European Testing Conference 2017 proudly introduces our four keynote speakers:</p>
 <p class="call-to-action"><a href="/news/Announcing-Keynote-Speakers">Read full article</a></p>
 </article>
 <article class="news">
 <h3><a href="/news/Extend-Call-For-Keynote">Extend call for keynote 2017 to 14.7.2016</a></h3>
 <p class="writer">28 Jun 2016 / Maaret Pyhäjärvi</p>
 </article>
 <article class="news">
 <h3><a href="/news/Calls-for-talks-2017">European Testing Conference 2017 is now searching for contents!</a></h3>
 <p class="writer">13 Jun 2016 / Maaret Pyhäjärvi</p>
 </article>
 <article class="news">
 <h3><a href="/news/see-you-in-Finland-2017">European Testing Conference 2017 goes to Finland</a></h3>
 <p class="writer">22 Feb 2016 / Maaret Pyhäjärvi</p>
 </article>
<p class="call-to-action"><a href="/news/">View archive</a></p>
</div>
<div class="col-md-6 col-xs-12">
	<hgroup>
	 <h2>Tweets<br />
	 <span></span>
	</h2>
	</hgroup>
	<div id="tweets">
	 <a class="twitter-timeline" href="https://twitter.com/EuroTestingConf" data-widget-id="624185830425366529">Tweets by @EuroTestingConf</a> <script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+"://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>
	</div>
</div>
</section>
<section id="why" class="col-md-12 main-content text-center ">
 <h2>Why</h2>
 <p><strong>European Testing Conference (ETC)</strong> mission is to build a practical conference that raises the bar in the testing world.</p>
<p>For our participants, we bring together testing from developer and tester perspectives in a practice-oriented package in a dialog culture, aiming for mutual learning of how testing can be done in practice.</p>
<p>For our speakers, we work against the trend of having to pay for speaking. We start with paying expenses of travel and accommodation. Depending on the success of the conference, we will also be paying a leveled honorarium. Eventually, we seek to pay for speaking and believe it attracts the best speakers to speak with us.</p>
<p>The conference is brought to you from the starting ideas of Adi Bolboacă (Developer from Romania) and Maaret Pyhäjärvi (Tester from Finland). We welcome the software community to contribute in organizing a people-oriented hands-on conference.</p>
<p><span class="interested">Interested? <a href="http://eepurl.com/brb31f" class="menu-special-button">Join our mailing list! </a></span></p>
</section>
<section id="blog" class="col-md-12 main-content text-center ">
 <h2>Our Blog</h2>
 <div id="blog-container" class="hidden blogs" data-resource="http://europeantestingconference.blogspot.com/feeds/posts/default?alt=json">
	<p>We blog about testing, and the event <a target="_blank" href="http://europeantestingconference.blogspot.com">here</a>. Below is an excerpt of the latest blog, and a list of the additional latest entries.
	<article class="blog-newest col-md-6" id="blog-latest">
		<header><h3></h3></header>
		<section></section>
		<footer></footer>
	</article>
	<article class="blog-rest col-md-6">
		<header><h3>Older blogs</h3></header>
		<ul id="blog-rest">
		</ul>
	&lt;/div&gt;
&lt;/div&gt;
</article></p></div>
</section>
<section id="directions" class="col-md-12 main-content text-center ">
 <h2>Directions: How to get there</h2>
 <h3 id="by-plane">By plane</h3>
<p>The nearest airport is Helsinki-Vantaa (HEL).</p>
</section>
<section id="volunteers" class="col-md-12 main-content text-center ">
 <h2>Organizers</h2>
 <ol class="volunteers-list">
 <li class="volunteer">
 <img src="https://pbs.twimg.com/profile_images/1299815722/Adi1.JPG" />
 <span class="name">Adi Bolboacă</span>
 <a href="https://twitter.com/adibolb">@adibolb</a>
 </li>
 <li class="volunteer">
 <img src="https://pbs.twimg.com/profile_images/1877737374/20110907at09-16-47.jpg" />
 <span class="name">Aki Salmi</span>
 <a href="https://twitter.com/rinkkasatiainen">@rinkkasatiainen</a>
 </li>
 <li class="volunteer">
 <img src="https://pbs.twimg.com/profile_images/621208613747576832/cdWmRNH7_400x400.jpg" />
 <span class="name">Alina Ionescu</span>
 <a href="https://twitter.com/al_innutza">@al_innutza</a>
 </li>
 <li class="volunteer">
 <img src="https://pbs.twimg.com/profile_images/752370366639661057/lw-P46W5.jpg" />
 <span class="name">Llewellyn Falco</span>
 <a href="https://twitter.com/LlewellynFalco">@LlewellynFalco</a>
 </li>
 <li class="volunteer">
 <img src="https://pbs.twimg.com/profile_images/629062251152961536/kumhZ5lm_400x400.jpg" />
 <span class="name">Maaret Pyhäjärvi</span>
 <a href="https://twitter.com/maaretp">@maaretp</a>
 </li>
</ol>
</section>
<section id="sponsor-block" class="col-md-12 main-content text-center ">
 <h2>Sponsors</h2>
 <ul id="sponsor-groups" class="list-unstyled">
<li class="active platinum-sponsor">
 <h3>Platinum sponsors </h3>
 <p class="sponsors">
 <h4>You? <a href="mailto:contact@europeantestingconference.eu">Contact us!</a></h4>
 </p>
</li>
<li class="next platinum-sponsor">
 <h3>Platinum sponsors</h3>
 <p class="sponsors">
 <h4>You? <a href="mailto:contact@europeantestingconference.eu">Contact us!</a></h4>
 </p>
</li>
<li class="gold-sponsor">
 <h3>Gold sponsors</h3>
 <p class="sponsors">
 <h4>You? <a href="mailto:contact@europeantestingconference.eu">Contact us!</a></h4>
 </p>
</li>
<li class="silver-sponsor">
 <h3>Silver sponsors</h3>
 <p class="sponsor">
 <h4>You? <a href="mailto:contact@europeantestingconference.eu">Contact us!</a></h4>
 </p>
</li>
<li class="next gold-sponsor">
 <h3>Partners</h3>
 <p class="sponsors">
 <h4>You? <a href="mailto:contact@europeantestingconference.eu">Contact us!</a></h4>
 </p>
</li>
</ul>
<h4>Interested in sponsoring European Testing Conference? <a href="/sponsorship/#sponsorship-portfolio">Check our offerings</a></h4>
</section>
</main>
<!-- build:js scripts/vendor.js -->
<!-- bower:js -->
<footer class="text-center container-fluid">
<p>This page is served by github pages.</p>
<p>Feel free to copy use the <a href="https://github.com/EuropeanTestingConference/europeantestingconference">structure</a> as basis for your own conference and/or web page.</p>
<p>You can reach organizers at <a href="mailto:contact@europeantestingconference.eu">contact@europeantestingconference.eu</a></p>
<h3>Platinum Sponsors</h3>
<div class="platinum-sponsor">
 <ul class="sponsors">
	 <li class="sponsor">
	 </li>
	 <li class="sponsor ">
	 </li>
 </ul>
 <ul class="sponsors">
	 <li class="sponsor ">
	 </li>
 </ul>
</div>
</footer>
<script src="/bower_components/jquery/dist/jquery.min.js"></script> 
<script src="/bower_components/modernizr/modernizr.js"></script>
<script src="/javascripts/play-sponsors.js"></script>
<script src="/javascripts/fix-navi.js"></script>
<script src="/javascripts/bootstrap.min.js"></script>
<script src="/javascripts/personas.js"></script>
<script src="/javascripts/blog.js"></script>
<script src="/javascripts/topics.js"></script>
 </body>
</html>
